specs/disciplina/educator/api/educator.yaml

```diff
@@ -1,515 +1,707 @@
       produces:
-      - "application/json"
+      - application/json
       parameters:
       - $ref: "#/parameters/SubmissionHash"
       responses:
         200:
-          description: "successful operation"
+          description: successful operation
+        400:
+          description: Submission hash value is invalid
+          schema:
+            $ref: "#/definitions/ApiResponse"
         404:
-          description: "Submission with given hash not found"
+          description: Submission with given hash not found
+          schema:
+            $ref: "#/definitions/ApiResponse"
 
   /grades:
     get:
       tags:
-      - "grades"
-      summary: "Get all grades"
-      description: "Gets a list of all grades performed by all students."
-      operationId: "getGrades"
+      - grades
+      summary: Get all grades
+      description: Gets a list of all grades performed by all students.
+      operationId: getGrades
       produces:
-      - "application/json"
+      - application/json
       responses:
         200:
-          description: "successful operation"
+          description: successful operation
           schema:
-            type: "array"
+            type: array
             items:
               $ref: "#/definitions/Grade"
     post:
       tags:
-      - "grades"
-      summary: "Post a new grade"
-      description: "Posts a new grade with a given body."
-      operationId: "postGrade"
+      - grades
+      summary: Post a new grade
+      description: Posts a new grade with a given body.
+      operationId: postGrade
       produces:
-      - "application/json"
+      - application/json
       consumes:
-      - "application/json"
+      - application/json
       parameters:
-      - in: "body"
-        name: "grade"
-        description: "A grade"
+      - in: body
+        name: grade
+        description: A grade
         required: true
         schema:
           $ref: "#/definitions/Grade"
       responses:
-        200:
-          description: "successful operation"
+        201:
+          description: successful operation
           schema:
             $ref: "#/definitions/Grade"
+        400:
+          description: Request body is invalid
+          schema:
+            $ref: "#/definitions/ApiResponse"
+
+  /assignments:
+    get:
+      tags:
+      - assignments
+      summary: Get all assignments
+      description: Gets a list of all student assignments. Filter parameters are used to specify specific course or student.
+      operationId: getAssignments
+      parameters:
+        - $ref: '#/parameters/CourseId'
+        - $ref: '#/parameters/StudentAddr'
+      produces:
+      - application/json
+      responses:
+        200:
+          description: successful operation
+          schema:
+            type: array
+            items:
+              $ref: "#/definitions/Assignment"
+    post:
+      tags:
+      - assignments
+      summary: Add assignment to a course
```

Comment:
I also think there should be an option to automatically assign the assignment to all the students on the course.

Reply:
Makes sense